tests/FakeItEasy.Specs/WrappingFakeSpecs.cs

```diff
@@ -0,0 +1,169 @@
+                .x(() => realObject.VoidMethodCalled.Should().BeTrue());
+        }
+
+        [Scenario]
+        public void VoidException(
+            Foo realObject,
+            IFoo wrapper,
+            Exception exception)
+        {
+            "Given a real object"
+                .x(() => realObject = new Foo());
+
+            "And a fake wrapping this object"
+                .x(() => wrapper = A.Fake<IFoo>(o => o.Wrapping(realObject)));
+
+            "When a void method is called on the wrapper"
+                .x(() => exception = Record.Exception(() => wrapper.VoidMethod(null)));
+
+            "Then the real object's method is called"
+                .x(() => realObject.VoidMethodCalled.Should().BeTrue());
+
+            "And the wrapper throws the exception thrown by the real object's method"
+                .x(() => exception.Should().BeAnExceptionOfType<ArgumentNullException>());
+        }
+
+        [Scenario]
+        public void OutAndRef(
+            Foo realObject,
+            IFoo wrapper,
+            int refParameter,
+            int outParameter)
+        {
+            "Given a real object"
+                .x(() => realObject = new Foo());
+
+            "And a fake wrapping this object"
+                .x(() => wrapper = A.Fake<IFoo>(o => o.Wrapping(realObject)));
+
+            "When a method with out and ref parameters is called is called on the wrapper"
+                .x(() => wrapper.OutAndRefMethod(ref refParameter, out outParameter));
+
+            "Then the real object's method is called"
+                .x(() => realObject.OutAndRefMethodCalled.Should().BeTrue());
+
+            "And the value of the ref parameter is modified"
+                .x(() => refParameter.Should().Be(1));
+
+            "And the value of the out parameter is set"
+                .x(() => outParameter.Should().Be(42));
+        }
+
+        public class Foo : IFoo
+        {
+            public bool NonVoidMethodCalled { get; private set; }
+
+            public bool VoidMethodCalled { get; private set; }
+
+            public bool OutAndRefMethodCalled { get; private set; }
+
+            public int NonVoidMethod(string parameter)
+            {
+                this.NonVoidMethodCalled = true;
+                if (parameter == null)
+                {
+                    throw new ArgumentNullException(nameof(parameter));
+                }
+
+                return parameter.Length;
+            }
+
+            public void VoidMethod(string parameter)
+            {
+                this.VoidMethodCalled = true;
+                if (parameter == null)
+                {
+                    throw new ArgumentNullException(nameof(parameter));
+                }
+            }
+
+            public void OutAndRefMethod(ref int refParameter, out int outParameter)
+            {
+                this.OutAndRefMethodCalled = true;
+                refParameter += 1;

```

Comment:
You shouldn't really do anything outside steps. The entire body of the test should really be inside the step delegates. Doing some stuff outside them will work, other stuff won't, so I think it's much better to keep everything inside them, e.g.

```C#
"When a method with out and ref parameters is called on the wrapper"
    .x(() =>
    {
        refParameter = 0;
        wrapper.OutAndRefMethod(ref refParameter, out outParameter);
    });
```

However, if you know that the wrapped object increments the ref param, then aren't the initial and final values fundamental to the scenario?